cupy/core/_routines_manipulation.pyx

```diff
@@ -435,7 +435,8 @@ cpdef array_split(ndarray ary, indices_or_sections, Py_ssize_t axis):
     if ary.size == 0:
         stride = 0
     for index in indices:
-        shape[axis] = index - prev
```

Comment:
I think that the way min & max are implemented in python can lead to multiple device synchronization issues here. 
Python could synchronize the device when reading the result of cupy.array.__lt__ and cupy.array.__gt__